source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -6892,7 +6902,14 @@ export class Container extends ContainerBase {
     protected internalToJSON(target: PropertyBag, context: SerializationContext) {
         super.internalToJSON(target, context);
 
-        context.serializeArray(target, this.getItemsCollectionPropertyName(), this._items);
+        const collectionPropertyName = this.getItemsCollectionPropertyName();
+
+        if ((this._items.length === 1) && (this._items[0].getElementSingletonBehavior() === ElementSingletonBehavior.Only)) {
```

Comment:
why are you checking if the length is 1?

Reply:
To confirm that this is only one element in the array

Reply:
We at least need a null check before calling `getElementSingletonBehavior()`. The length check ensures that there is only one element in the provided array.

We could strengthen the singleton behavior here so that the edge case we avoid (ex: `_items = ["carousel", "textblock, ...]` never occurs